Main PR

HarryC05 opened the pull request:
- Adds a new release-note post type
- Adds a rest point to create a new release note
- Adds a markdown parser block
- Add the ability to create a release on GitHub and automatically create a release note with circleCi and the rest point
- Adds colour indicators to check if a release is a pre-release or a full release

Video: http://bigbite.im/v/JQ3hb8

HarryC05 committed:
chore: inital commit

HarryC05 committed:
feat: added release notes post type

HarryC05 committed:
fix: global variable names

HarryC05 committed:
feat: added a simple template for the release notes post type

HarryC05 committed:
feat: added the widget to the dashboard

HarryC05 committed:
feat: added the archive page

HarryC05 committed:
feat: added the admin bar item

HarryC05 committed:
feat: added the version number field to the post and cleaned up files

HarryC05 committed:
fix: changed the header place holders for content

HarryC05 committed:
feat: added video block support

HarryC05 committed:
feat: added a release date meta

HarryC05 committed:
fix: added a label to the date picker input

HarryC05 committed:
chore: moved all changes made in the other repo

Moved all the changes made in the jasonsdemo repo over

HarryC05 committed:
feat: add settings page

HarryC05 committed:
fix: re worded settings names

HarryC05 committed:
feat: added support for the slack API

Adds support for headers, paragraphs, lists, and images

HarryC05 committed:
feat: added a header section to the slack message

HarryC05 committed:
chore: removed version section from slack message

HarryC05 committed:
feat: added support for links and rich text formatting

HarryC05 committed:
feat: created richtext and link formatters

HarryC05 committed:
chore: add doc comments

HarryC05 committed:
chore: added doc comments

HarryC05 committed:
style: added space before/after brackets

HarryC05 committed:
chore: removed unused file

HarryC05 committed:
feat: added the ability to use post meta in the slack message

HarryC05 committed:
feat: added different header for a pre-release

HarryC05 committed:
fix: updated the namespace to use camelcaps

HarryC05 committed:
chore: added editor config file

HarryC05 committed:
chore: replaced indentation with tabs

HarryC05 committed:
chore: added domain to the translation strings

HarryC05 committed:
chore: added more translations and fixed spacing

HarryC05 committed:
chore: doc comment cleanup and function clean up

HarryC05 committed:
chore: replaced double quotes, fixed spacing, added translations

HarryC05 committed:
chore: updated dic comments

HarryC05 committed:
chore: separated the foreach loop into its own func

HarryC05 committed:
chore: reduced the complexity of the format func

HarryC05 committed:
chore: removed useless validation function

HarryC05 committed:
fix: update phpcs ruleset ref

HarryC05 committed:
chore: added comments to empty files

HarryC05 committed:
style: phpcs changes

HarryC05 committed:
style: phpcs

HarryC05 committed:
style: phpcs

HarryC05 committed:
style: split all if statements into their own func

HarryC05 committed:
feat: now using the rich text block for the regular text

HarryC05 committed:
feat: added rich text lists

HarryC05 committed:
style: phpcs

HarryC05 committed:
fix: fix image links and alt text for slack notification

HarryC05 committed:
fix: moved class properties and used sprintf for strings

HarryC05 committed:
fix: added default value to the settings

HarryC05 committed:
fix: default setting arr

HarryC05 committed:
style: phpcs

ampersarnie reviewed (commented):
@HarryC05 How much of this is cross-over with #2? Seeing a fair amount of the same or similar issues.

HarryC05 committed:
Merge pull request #2 from bigbite/feat/slack-integration

Feat: slack integration

HarryC05 committed:
chore: removed class name space

HarryC05 committed:
fix: removed unused function

HarryC05 committed:
fix: updated heading regex pattern

ampersarnie reviewed (changes requested):
Just this last thing that I've noticed here. I think you should be good to go after this.

HarryC05 committed:
Update class-register-settings.php

HarryC05 committed:
chore: composer update

HarryC05 committed:
Merge branch 'main' of github.com:bigbite/release-notes

HarryC05 committed:
fix: changed __e() to _e()

HarryC05 committed:
fix: allow classes on elements

HarryC05 committed:
fix: remove dives from the elements list

HarryC05 committed:
Updated config.yml

HarryC05 committed:
Update composer.json

HarryC05 committed:
Update .deployignore

HarryC05 committed:
Update plugin.php

HarryC05 committed:
Update README.md

HarryC05 committed:
chore: changed plugin type to myplugin

HarryC05 committed:
updated plugin and readme file

HarryC05 committed:
revert README.md

HarryC05 committed:
revert composer.json

HarryC05 committed:
revert plugin.php

HarryC05 committed:
revert composer.json

HarryC05 committed:
chore: change plugin type to client mu

HarryC05 committed:
chore: update version numbers

HarryC05 committed:
revert

HarryC05 committed:
chore: changed to mu plugin type

HarryC05 committed:
chore: update version

HarryC05 committed:
revert

HarryC05 committed:
chore: composer update

g-elwell committed:
chore: install cypress and wp-cypress

g-elwell committed:
chore: install and configure cucumber preprocessor, add example test

HarryC05 committed:
Chore: added instructions to README

Added instructions for slack integration, github integration and remotely creating posts

g-elwell committed:
chore: add gherkin lint

g-elwell committed:
feat: remove unused seeders, seed release notes, rewrite scenarios

g-elwell committed:
Merge pull request #3 from bigbite/chore/cypress-setup

Chore: Initial Cypress setup

Commit:
test: initial creation of a release note for the test

Commit:
test: implentation of successfully publishing the release note

Commit:
test: assertions added to test

Commit:
refactor: refactored the feature and test files along with the naming of the files.

Commit:
chore: upgrade wp-cypress to current version

Commit:
chore: removed all reference of cucumber and gherkin from the plugin and refactored the tests to use Cypress syntax

jaymcp committed:
chore: add CODEOWNERS file

johnmunster committed:
Merge pull request #5 from bigbite/feature/release-notes-tests

[4] Upgrade wp-cypress, remove cucumber, gherkin linter and refactor tests back to Cypress syntax

Commit:
feature: non admin users test added

Commit:
fix: removed the page template test as it was not complete

Commit:
featre: Added some code that stops xhr requests from showing when running tests

Commit:
feature: creation of all users can access releae notes page

Commit:
fix: typo fix and changed object of varibles to an array

Commit:
fix: refactored the test to loop over all users to run as one test

Commit:
feature: new test for users seeing the release notes widget in the dashboard and a seeder that seeds several release notes

Commit:
fix: fixed a typo found in PR review

johnmunster committed:
Update cypress/e2e/access-to-release-notes-page.cy.js

Co-authored-by: Gareth Elwell <41474928+g-elwell@users.noreply.github.com>

johnmunster committed:
Merge pull request #10 from bigbite/feature/access-to-release-notes-page

[4] All users can access release notes page test

johnmunster committed:
Merge pull request #9 from bigbite/feature/remove-xhr-requests

[4] Addition of some code the stops xhr requests showing in cypress console log

johnmunster committed:
Merge pull request #8 from bigbite/feature/release-notes-tests

[4] addition of non admin users privileges test

Commit:
add: New test that varifies that is possible to delete release notes:
q

Commit:
fix: adjusted the selector in the test to grab the most recent release note

Commit:
fix: added an extra assertion to the version in the top toolbar and changed the name of the file to be more specific to the current release note as the new assertion should only occur when the most recent release note is delete

johnmunster committed:
Merge pull request #11 from bigbite/feature/users-see-release-notes-in-dashboard

[4] new test for users seeing the release notes widget in dashboard

Commit:
test: refactored the test to make it more about the version control than the deletion of a version

johnmunster committed:
Merge branch 'main' into feature/delete-release-note

Commit:
test: filename, test description and slight refactor of code so that assertion is only finding one element

johnmunster committed:
Merge pull request #13 from bigbite/feature/delete-release-note

[4] Version control change after release note deletion

jaymcp committed:
Merge pull request #7 from bigbite/chore/codeowner

chore: add CODEOWNERS file

jaymcp committed:
docs: add GPLv2 licence

jonnywatersbb committed:
Merge pull request #20 from bigbite/docs/licence

[#19] - Add GPLv2 licence
